estimate-web/src/components/UserStoryCommentEditor.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from "react";
+import { PropTypes } from "prop-types";
+
+import {
+    TextField, Persona, PersonaSize, PrimaryButton, Spinner
+} from "office-ui-fabric-react";
+
+import "./UserStoryCommentEditor.less";
+import { connect } from "react-redux";
+import userShape from "../reducers/models/userShape";
+import { requestWorkItemAddComment } from "../actions/devops";
+import workItemShape from "../reducers/models/workItemShape";
+
+class UserStoryCommentEditor extends Component {
+    state = {
+        comment: "",
+        workItem: undefined
```

Comment:
It's confusing that there's both state.workItem and props.workItem

Reply:
I left a comment on the state variable, I hope this clarifies why it's there.
We need to keep the workItem in the state to detect when it changes, so we can reset the comment. I can't use `componentDidUpdate` as doing a `setState({comment: ""})` in there would cause an intial `render` call (also throws an eslint-warning).

Let's see, when I refactor the editor to not always be visible, maybe this code will become obsolete as well.